dali/test/python/test_external_source_parallel.py

```diff
@@ -59,21 +102,23 @@ def check_source_build(source):
 
 
 def test_wrong_source():
-    common_msg = "External Source in parallel mode (when `parallel=True`) accepts as `source` only *. Got {} instead"
+    callable_msg = ("Callable passed to External Source in parallel mode (when `parallel=True`) "
+        "must accept exactly one argument*. Got {} instead.")
+    batch_required_msg = "Parallel external source with {} must be run in a batch mode"
     expected_error_msgs = [
-        common_msg.format("a callable that does not accept arguments"),
-        "External source callback must be a callable with 0 or 1 argument",
-        common_msg.format("an iterable"),
-        common_msg.format("a generator function")]
+        (TypeError, callable_msg.format("a callable that does not accept arguments")),
```

Comment:
Maybe it is better to move this error messages next to `disallowed_sources` to make sure they match and are up to date.